diff --git a/hugo/internal/hugo_site.bzl b/hugo/internal/hugo_site.bzl
index 33a82a8..b1e64c8 100644
--- a/hugo/internal/hugo_site.bzl
+++ b/hugo/internal/hugo_site.bzl
@@ -46,7 +46,8 @@ def _hugo_site_impl(ctx):
     hugo_args = []
 
     # Copy the config file into place
-    config_file = ctx.actions.declare_file(ctx.file.config.basename)
+    config_name = "config.toml"
+    config_file = ctx.actions.declare_file(config_name)
     ctx.actions.run_shell(
         inputs = [ctx.file.config],
         outputs = [config_file],
@@ -201,7 +202,7 @@ hugo_site = rule(
     implementation = _hugo_site_impl,
 )
 
-_SERVE_SCRIPT_PREFIX = """#!/bin/bash
+_SERVE_SCRIPT_PREFIX = """#!/usr/bin/env bash
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 
 trap exit_gracefully SIGINT
