diff --git a/haskell/haddock.bzl b/haskell/haddock.bzl
index cec97792..d0ad7033 100644
--- a/haskell/haddock.bzl
+++ b/haskell/haddock.bzl
@@ -278,19 +278,23 @@ def _haskell_doc_rule_impl(ctx):
         )
 
     # Do one more Haddock call to generate the unified index
-
     index_root_raw = paths.join(doc_root_raw, "index")
     index_root = ctx.actions.declare_directory(index_root_raw)
 
     args = ctx.actions.args()
+
     args.add_all([
         "-o",
         index_root.path,
         "--title={0}".format(ctx.attr.name),
-        "--gen-index",
-        "--gen-contents",
     ])
 
+    if ctx.attr.index:
+        args.add_all([
+            "--gen-index",
+            "--gen-contents",
+        ])
+
     if ctx.attr.index_transitive_deps:
         # Include all packages in the unified index.
         for package_id in html_dict_copied:
@@ -344,6 +348,10 @@ haskell_doc = rule(
             ],
             doc = "List of Haskell libraries to generate documentation for.",
         ),
+        "index": attr.bool(
+            default = True,
+            doc = "Whether to use --gen-index to generate a combined index.",
+        ),
         "index_transitive_deps": attr.bool(
             default = False,
             doc = "Whether to include documentation of transitive dependencies in index.",
